### DISCUSSION TITLE
using Custom Dataset

### DISCUSSION THREAD

**Discussion by Divya216** [online-ml/river#1143] [Q&A] [answered]
I am using my custom dataset for my project - PhishingData.csv which has 11055 rows and 32 features  and have converted it into dictionaries to evaluate using the below code:

dataset = [(x, x.pop("Result")) for x in  df.to_dict(orient = "records")]

I would like to apply some models on my list dataset like the below from your River API reference

model = anomaly.QuantileFilter(
...     anomaly.OneClassSVM(nu=0.2),
...     q=0.995
... )

>>> auc = metrics.ROCAUC()

>>> for x, y in **dataset**.take(2500):
...     score = model.score_one(x)
...     is_anomaly = model.classify(score)
...     model = model.learn_one(x)
...     auc = auc.update(y, is_anomaly)

>>> auc

In the following line if I replace  datasets.CreditCard() with just my dataset it gives me the below error:

AttributeError: 'list' object has no attribute 'take'

How can i handle this list ?? 

![IMG_2571](https://user-images.githubusercontent.com/67040404/211130593-1d226102-26f2-4153-b2bd-d51090efd750.jpg)

**Comment by MaxHalford** [comment, marked as answer]
Hey hey 👋

So the datasets in the `datasets` module are `Dataset` instances. Meaning they have a `.take(k)` method. But your dataset is a list, so you don't have a `.take(k)` method. What you can do is simply slice your list:

```py
for x, y in dataset[:2500]:
    ...
```

You could also create a River dataset by inheriting from the `Dataset` base class, but that's not as straightforward.

**Reply by Divya216** [reply]
Slicing Worked !! Thanks a ton 👍
